### PR TITLE
[NetAppFiles] Anf 10301 net app files update swagger rest api to 2020 12 01

### PR DESCRIPTION
**Pull request by audunn**
This updates the Azure NetAppFiles to api version 2020-12-01

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ArcturusZhang** [comment]
These changes look good to me, please wait for ARM's feedback

**Comment by audunn** [comment]
@ArcturusZhang The pipeline is not completing all checks do we have any idea what it the issue.
`Swagger ApiDocPreview In progress — Swagger ApiDocPreview in progress`

**Comment by ArcturusZhang** [comment]
> @ArcturusZhang The pipeline is not completing all checks do we have any idea what it the issue.
> `Swagger ApiDocPreview In progress — Swagger ApiDocPreview in progress`

This is a known issue in ApiDocPreview, we could just ignore it for now.

**Comment by chiragg4u** [comment]
Please go through the contribution checklist added as part of the review and mark them completed. This is a requirement before review can be continued.

**Comment by audunn** [comment]
> Please go through the contribution checklist added as part of the review and mark them completed. This is a requirement before review can be continued.

@chiragg4u  done

**Comment by audunn** [comment]
note removed name property from snapshotpolicyProperties it is a violation
https://portal.microsofticm.com/imp/v3/incidents/details/209491961/home

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-12-01/examples/Accounts_CreateOrUpdate.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{
+  "parameters": {
+    "subscriptionId": "D633CC2E-722B-4AE1-B636-BBD9E4C60ED9",
+    "resourceGroupName": "myRG",
+    "accountName": "account1",
+    "api-version": "2020-11-01",
```
Comment (chiragg4u) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-12-01/examples/Accounts_CreateOrUpdate.json:6]: >"api-version": "2020-11-01", [](start = 4, length = 28)

Please fix API versions across all the files.

**Comment by chiragg4u** [comment]
>       "properties": {

Why are we removing the name? 

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-12-01/netapp.json:4211 in aaf12be. [](commit_id = aaf12be72ceec9f965b46c86657bdd6848eaca5e, deletion_comment = False)

**Comment by audunn** [comment]
> > ```
> >   "properties": {
> > ```
> 
> Why are we removing the name?
> 
> Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-12-01/netapp.json:4211 in [aaf12be](https://github.com/Azure/azure-rest-api-specs/commit/aaf12be72ceec9f965b46c86657bdd6848eaca5e). [](commit_id = [aaf12be](https://github.com/Azure/azure-rest-api-specs/commit/aaf12be72ceec9f965b46c86657bdd6848eaca5e), deletion_comment = False)

This was causing flags being raised, apparently a violation 

_"The resource properties shouldn't contain part of the resource definition, elements like id, subscription id, resource group name, the only valid ​resource property in the properties JToken is Sku.
https://armwiki.azurewebsites.net/api_contracts/RPCstrongvalidation.html 
"_

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-12-01/examples/Accounts_CreateOrUpdate.json (hunk 1 above)
Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-12-01/examples/Accounts_CreateOrUpdate.json]: sorry about that, done

**Review by j5lim — APPROVED** [review]
:shipit:

**Review by ArcturusZhang — APPROVED** [review]

**Comment by ArcturusZhang** [comment]
Please let me know when this is ready to merge

**Comment by audunn** [comment]
@ArcturusZhang it is ready to go thank you

**Review by msyyc — APPROVED** [review]

**Comment by msyyc** [comment]
For python, the breaking change is ok

**Review by msyyc — APPROVED** [review]

**Pull request merged**